packages/starknet-snap/src/index.tsx

```diff
@@ -341,3 +355,112 @@ export const onUpdate: OnUpdateHandler = async () => {
 export const onHomePage: OnHomePageHandler = async () => {
   return await homePageController.execute();
 };
+
+/**
+ * Handle incoming user events coming from the MetaMask clients open interfaces.
+ *
+ * @param params - The event parameters.
+ * @param params.id - The Snap interface ID where the event was fired.
+ * @param params.event - The event object containing the event type, name, and
+ * value.
+ * @param params.context
+ * @see https://docs.metamask.io/snaps/reference/exports/#onuserinput
+ */
+export const onUserInput: OnUserInputHandler = async ({
+  id,
+  event,
+  context,
+}: {
+  id: string;
+  event: UserInputEvent;
+  context: InterfaceContext | null;
+}): Promise<void> => {
+  const generateEventKey = (type: UserInputEventType, name: string) =>
+    `${type}_${name}`;
+  const eventKey = generateEventKey(event.type, event.name ?? '');
+
+  await updateInterface(
+    id,
+    <Box alignment="space-between" center={true}>
+      <Heading>please wait...</Heading>
+      <Spinner />
+    </Box>,
+  );
+
+  switch (eventKey) {
+    case generateEventKey(
+      UserInputEventType.InputChangeEvent,
+      'feeTokenSelector',
+    ): {
+      const networkStateMgr = new NetworkStateManager();
+      const network = await networkStateMgr.getCurrentNetwork();
```

Comment:
we should use the network from the request, to simplify you can even pass the network object in the request
so you dont need to have chainId & chain name from the transactionRequest